hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/mapred/RetriableFileCopyCommand.java

```diff
@@ -129,16 +129,15 @@ private long doCopy(CopyListingFileStatus source, Path target,
       throws IOException {
     LOG.info("Copying {} to {}", source.getPath(), target);
 
+    final Configuration configuration = context.getConfiguration();
```

Comment:
If you just get the schema in getTempFile() from the path, no need to pass in FileSystem, so no need to move this up.